keep-ui/app/(health)/health/page.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
+
+import ProvidersTiles from "@/app/(keep)/providers/providers-tiles";
+import React, { useEffect, useState } from "react";
+import { defaultProvider, Provider } from "@/app/(keep)/providers/providers";
+import { useProvidersWithHealthCheck } from "@/utils/hooks/useProviders";
+import Loading from "@/app/(keep)/loading";
+
+const useFetchProviders = () => {
+  const [providers, setProviders] = useState<Provider[]>([]);
+  const { data, error, mutate } = useProvidersWithHealthCheck();
+
+  if (error) {
+    throw error;
+  }
+
+  const isLocalhost: boolean = true;
+
+  useEffect(() => {
+    if (data) {
+      const fetchedProviders = data.providers
+        .filter((provider: Provider) => {
+          return provider.health;
+        })
+        .map((provider) => ({
+          ...defaultProvider,
+          ...provider,
+          id: provider.type,
+          installed: provider.installed ?? false,
+          health: provider.health,
+        }));
+
+      setProviders(fetchedProviders);
+    }
+  }, [data]);
+
+  return {
+    providers,
+    error,
+    isLocalhost,
+    mutate,
+  };
+};
+
+export default function ProviderHealthPage () {
+  const { providers, isLocalhost, mutate } = useFetchProviders();
+
+  if (!providers || providers.length <= 0) {
```

Comment:
It's not possible to have providers.length equal to 0? There should always be some provider?
If I'm not wrong, in the case when providers array is empty it will be infinite Loading.

Reply:
No, since we already have at least one provider which supports health checks it will be always more then zero